src/book/03-guides/01-main-codebase/01-getting-started.mdx

```diff
@@ -26,15 +30,19 @@ There is a guide to using Git in NUbots [here](/guides/general/git).
 
 Docker is a container system that allows you to install development tools and run builds in an isolated environment that won't affect the rest of your system. We use it to set up the NUbots build system and run builds and binaries.
 
-#### Ubuntu
+You will need to perform the steps correctly to set up Docker for your operating system to complete the example.
+
+<details>
+
+<summary>Ubuntu</summary>
 
 1. Add Docker's stable repository and update the apt package index.
 
    ```bash
    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
    sudo add-apt-repository \
      "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
-     bionic \
+     focal \
      stable"
    sudo add-apt-repository universe
    sudo apt update
```

Comment:
I think these two update commands should be added before the install commands (also, couldn't the install commands be merged into one command?), because I think mint users will run these commands, then it won't work, then they'll run the mint command, then they'll run the install commands, which won't work because these commands were ran before the special mint command.
I can see myself falling victim to this, because usually I am an idiot who doesn't read the next steps before running the current ones lmao

Reply:
Do you mean after? I've split up the commands into different steps, is it better?
On the one command thing, is that with `sudo add-apt-repository universe && sudo apt update` do you mean or something else? I'd rather not concatenate commands incase errors come up between commands.

Reply:
I think where you have put step 2:
```sh
sudo add-apt-repository universe
sudo apt update
``` 
is perfect now.

The commands I was thinking could be merged would be 
```sh
sudo apt-get install docker-ce docker-ce-cli containerd.io
sudo apt-get install python3 python3-pip
```
becomes:
```sh
# Pretty sure this would work?
sudo apt-get install docker-ce docker-ce-cli containerd.io python3 python3-pip
```

Sidenote: that step says "Install the lastest version of Docker", then it proceeds to install docker, docker's dependencies (fair), and python too. Could perhaps just change it to saying something which encompasses both?

Reply:
Looks like they have changed the steps to install Docker, so we'll need to add a few commands. I tried to test the combined command in a fresh ubuntu container, and now the steps we have listed won't work because we haven't added the GPG key for the repo...

https://docs.docker.com/engine/install/ubuntu/

Gonna have to add a couple of steps.

Something that's really nice though, is that they have added a command to the part where you change between focal and bionic so that we don't have to hardcode the version at all any more. Might not need divergent mint and ubuntu steps now.

Reply:
On second thought, I reckon we could just change the guide to say "Go to Docker(link) and follow the steps to install it." at this stage (maybe with a comment saying that mint users should just follow the ubuntu steps).
That would make that `apt-get install <packages>` just `sudo apt-get install python3 python3-pip`

Reply:
If it's `$(lsb_release -cs)`, I think that's always been a thing, it just doesn't work on Linux Mint.
Are the extra steps just 1 and 2 of 'SET UP THE REPOSITORY'?
We could add a link, kind of like an 'in case this doesn't work check if the instructions have changed' kind of thing. But there's a lot of extra stuff on there, and multiple ways to do things... so probs good to still have instructions here.

Reply:
The extra steps involve installing 5 more packages, then curling and adding the docker GPG key, checking the fingerprint to make sure you did those properly, then the steps we already have. I am not sure how long that fingerprint would be the same, so I am sceptical that we might have to update that regularly

Reply:
Yeah maybe we should just link to it then. Then specify what section to do. And add any notes, like how to find what Ubuntu distro your Linux Mint version is.